agent/chaldea/__init__.py

```
@@ -15,11 +16,138 @@
+
+def _load_local_file(source: str) -> Optional[dict]:
+    """尝试把输入当作本地文件路径/文件名加载 BattleShareData。
+
+    支持:
+    - 绝对/相对路径: G:/xxx/team.json / ./team.json
+    - 纯文件名: 自动在 CACHE_DIR (config/Battle) 中查找
+    - .json 后缀可省略
+    返回加载成功的 dict, 不是文件输入返回 None。
+    """
+    s = source.strip()
+    if not s or re.match(r'^https?://', s) or s.isdigit() or s.startswith("G") and len(s) < 100 and not s.lower().endswith(".json"):
```

Ruff (0.16.2)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)